### PR TITLE
df.melt() column name 'variable'

### LINKED ISSUE #37716: df.melt() overwriting column named 'variable'
when melting df using a column named 'variable' as an id_var.
resultant frame has original 'variable' column overwritten with the newly melted 'variable' column.
would suggest checking for presence of 'variable' column and appending '_i' to newly created columns.

[example_melt_overwrite.zip](https://github.com/pandas-dev/pandas/files/5510886/example_melt_overwrite.zip)

foo = pd.read_pickle('example_melt_overwrite.pkl')
foo.melt(id_vars=['DateTime','variable'])

### PR DESCRIPTION
**Pull request by micyhkim**
- [x] closes #37716
- [x] tests added / passed
- [x] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

This is my first time contributing to the open source project. Although I didn't make a significant change, I saw a bug report and asked to just raise a warning instead of fixing it. That's what I did and returned original DataFrame instead of returning wrong frame.

### CODE REVIEW

**Review by arw2019 — COMMENTED** [review]
Thanks @micyhkim for the PR!

We'll need a test for this. This will go in pandas/tests - look for similar tests to figure out where to place it and the patterns. Use `assert_produces_warning` from `pandas._testing` to check that the warning is raised.

Let us know if you have any questions!

File: pandas/core/reshape/melt.py [hunk 1]
```diff
@@ -108,6 +118,7 @@ def melt(
             var_name = [
                 frame.columns.name if frame.columns.name is not None else "variable"
             ]
+
```
Comment (mroeschke) [inline pandas/core/reshape/melt.py:121]: Can you undo this white space?

File: pandas/core/reshape/melt.py [hunk 2]
```diff
@@ -39,6 +39,16 @@ def melt(
     else:
         cols = list(frame.columns)
 
+    if 'variable' in frame.columns:
+        warnings.warn(
+            "Using column name 'variable' might cause "
+            "overwriting in resulting Dataframe. "
+            "Please don't use 'variable' in column name",
+            UserWarning,
```
Comment (mroeschke) [inline pandas/core/reshape/melt.py:47]: This should raise a `FutureWarning`

File: pandas/core/reshape/melt.py [hunk 3]
```diff
@@ -39,6 +39,16 @@ def melt(
     else:
         cols = list(frame.columns)
 
+    if 'variable' in frame.columns:
+        warnings.warn(
+            "Using column name 'variable' might cause "
```
Comment (mroeschke) [inline pandas/core/reshape/melt.py:44]: I think instead we should mention that a `ValueError` will raise if `frame.columns.name is None` in the future.

So this warning should be placed below by:

```
            var_name = [
                frame.columns.name if frame.columns.name is not None else "variable"
            ]
```

**Comment by jreback** [comment]
closing as stale. if you want to continue, pls ping and can re-open.

**Pull request closed without merging**